app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,27 @@
+class CommentsController < ApplicationController
+  def new
+    @user = current_user
+    @post = Post.find(params[:id])
+    @comment = Comment.new
+  end
+
+  def create
+    @user = current_user
+    @post = Post.find(params[:id])
+    @comment = Comment.new(comment_params)
+    @comment.author = current_user
```

Comment:
- [ ] After creating `@user` we are not using it on line 12, kindly use `@comment.author = @user`. so we dont  have to call again current_user method.